samples/sample-kotlin-android/src/main/kotlin/com/rudderstack/android/sampleapp/analytics/RudderAnalyticsUtils.kt

```diff
@@ -22,34 +22,43 @@ object RudderAnalyticsUtils {
     private var _rudderReporter: RudderReporter? = null
 
     fun initialize(application: Application, listener: InitializationListener? = null) {
-        _rudderAnalytics = createInstance(
-            writeKey = WRITE_KEY,
+        _rudderAnalytics = createInstance(writeKey = WRITE_KEY,
+            application = application,
+            jsonAdapter = GsonAdapter(),
+
             initializationListener = { success, message ->
                 listener?.onAnalyticsInitialized(PRIMARY_INSTANCE_NAME, success, message)
             },
-            configuration = ConfigurationAndroid(
-                application = application,
-                GsonAdapter(),
-                dataPlaneUrl = DATA_PLANE_URL,
-                controlPlaneUrl = CONTROL_PLANE_URL,
-                trackLifecycleEvents = true,
-                recordScreenViews = true,
-            )
-        )
-        _rudderAnalyticsSecondary = createInstance(
-            writeKey = WRITE_KEY_SECONDARY,
+            configurationInitializer = {
+//                dataPlaneUrl = DATA_PLANE_URL,
+//                controlPlaneUrl = CONTROL_PLANE_URL,
+//                recordScreenViews = true,
+                copy(
```

Comment:
Please would you like to explain more about this `copy` method and that callback added here? Frankly, it does not look much user-friendly.